make-release.sh

```diff
@@ -11,147 +11,221 @@
     git commit -asm "${COMMIT_MSG}"
   fi
   git pull origin "${BUMP_BRANCH}"
 
-  set +e
-  PUSH_TRY="$(git push origin "${BUMP_BRANCH}")"
+  if [[ -z "$DRY_RUN" ]]; then
+    set +e
+    PUSH_TRY="$(git push origin "${BUMP_BRANCH}")"
+    PUSH_TRY_EXIT_CODE=$?
+    set -e
+  else
+    PUSH_TRY="protected branch hook declined"
+  fi
   # shellcheck disable=SC2181
-  if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then
+  if [[ "${PUSH_TRY_EXIT_CODE}" -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then
     PR_BRANCH=pr-${BUMP_BRANCH}-to-${NEXT_VERSION}
     # create pull request for the main branch branch, as branch is restricted
     git branch "${PR_BRANCH}"
     git checkout "${PR_BRANCH}"
-    git pull origin "${PR_BRANCH}"
-    git push origin "${PR_BRANCH}"
+    git pull origin "${PR_BRANCH}" || true
+    $DRY_RUN git push origin "${PR_BRANCH}"
     lastCommitComment="$(git log -1 --pretty=%B)"
-    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
+    $DRY_RUN hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
   fi
-  set -e
   git checkout "${CURRENT_BRANCH}"
 }
 
-usage ()
-{
-  echo "Usage: $0 --version [VERSION TO RELEASE]"
-  echo "Example: $0 --version v0.1.0"; echo
-}
+prerelease() {
+  echo "[INFO] Starting prerelease procedure"
+  # derive bugfix branch from version
+  BRANCH=${VERSION#v}
+  BRANCH=${BRANCH%.*}.x
 
-if [[ ! ${VERSION} ]]; then
-  usage
-  exit 1
-fi
+  echo "[INFO] Creating ${BRANCH} from ${MAIN_BRANCH}"
+  git checkout ${MAIN_BRANCH}
+  git branch "${BRANCH}" || git checkout "${BRANCH}"
+  sed -i version/version.go -r -e "s#(Version = \")(v?[0-9.]+-rc)(\")#\1${VERSION}\3#g"
+  if [[ ! -z $(git status -s) ]]; then # dirty
+    COMMIT_MSG="[prerelease] Remove rc from version"
+    git commit -asm "${COMMIT_MSG}"
+  fi
+  $DRY_RUN git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout "${BRANCH}"
 
+  # change VERSION file + commit change into ${BASEBRANCH} branch
+  # bump the y digit, if it is a major release
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=0.1.x, get BASE=0, NEXT=2
+  NEXT_VERSION_Y="${BASE}.${NEXT}.0"
+  bump_version "${NEXT_VERSION_Y}" "${MAIN_BRANCH}"
+  echo "Prerelease is done"
+}
 
-# derive bugfix branch from version
-BRANCH=${VERSION#v}
-BRANCH=${BRANCH%.*}.x
+release() {
+  echo "[INFO] Starting Release procedure"
+  # derive bugfix branch from version
+  BRANCH=${VERSION#v}
+  BRANCH=${BRANCH%.*}.x
 
-# if doing a .0 release, use main branch; if doing a .z release, use $BRANCH
-if [[ ${VERSION} == *".0" ]]; then
-  BASEBRANCH="${MAIN_BRANCH}"
-else
   BASEBRANCH="${BRANCH}"
-fi
 
-# work in tmp dir
-if [[ $TMP ]] && [[ -d $TMP ]]; then
-  pushd "$TMP" > /dev/null || exit 1
+  git remote show origin
```

Comment:
Is this a debugging message?

Reply:
Blame me for moving stuff without paying enough attention.
I suppose so, will remove it until we don't see need in it.